File: predator-hibernate-jpa/src/test/java/io/micronaut/data/hibernate/AuthorRepository.java
package io.micronaut.data.hibernate;

import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.CrudRepository;

import javax.transaction.Transactional;
import java.util.List;

@Repository
@Transactional
public interface AuthorRepository extends CrudRepository<Author, Long> {

    Author findByName(String name);

    Author findByBooksTitle(String title);
}

File: doc-examples/hibernate-example-kotlin-ksp/src/main/kotlin/example/AuthorRepository.kt
package example

import io.micronaut.data.annotation.Join
import io.micronaut.data.annotation.Repository
import io.micronaut.data.repository.CrudRepository
import io.micronaut.data.repository.jpa.JpaSpecificationExecutor

@Repository
//@Join(value = "genres", type = Join.Type.LEFT_FETCH)
interface AuthorRepository : CrudRepository<Author, Long>, JpaSpecificationExecutor<Author>
